akka-stream/src/main/scala/akka/stream/remote/scaladsl/SinkRef.scala

```diff
@@ -55,22 +55,22 @@ final class SinkRefTargetSource[T] extends GraphStageWithMaterializedValue[Sourc
     val promise = Promise[SinkRef[T]]()
 
     val logic = new GraphStageLogic(shape) with StageLogging with OutHandler {
-
-      //      var receiverRef: ActorRef = _
+      private[this] lazy val streamRefsMaster = StreamRefsMaster(ActorMaterializerHelper.downcast(materializer).system)
 
       private[this] var self: GraphStageLogic.StageActor = _
-      private[this] implicit def selfSender = self.ref
+      private[this] lazy val selfActorName = streamRefsMaster.nextSinkRefTargetSourceName()
+      private[this] implicit def selfSender: ActorRef = self.ref
 
       val initialDemand = 4L // TODO get from config as well as attributes
 
       var localCumulativeDemand = 0L
       var remotePartner: ActorRef = _
 
       override def preStart(): Unit = {
-        self = getStageActor(initialReceive)
-        //        val master = StreamRefsMaster(ActorMaterializerHelper.downcast(materializer).system)
+        self = getStageActor(initialReceive, name = selfActorName)
```

Comment:
The names are like ` Allocated receiver: Actor[akka://RemoteSystem/user/actor/StreamSupervisor-1/$$a-SinkRefTargetSource-0#63380146]`

Reply:
So it's visible it's a function ref, but contains the given name hm. Perhaps worth re-thinking how the names should work. But it helps a lot to have the named part in them already